storybook/render_macros.py

```diff
@@ -0,0 +1,109 @@
+import os
+
+from jinja2 import Environment, FileSystemLoader, select_autoescape
+
+BASE_DIR = os.path.dirname(__file__)
+MACROS_DIR = os.path.join(BASE_DIR, "..", "app", "templates", "main", "macros")
+OUTPUT_DIR = os.path.join(BASE_DIR, "html")
+
+# Define macro render tasks.
+# Each task can either define:
+# - A single variant with "context" and "output", OR
+# - Multiple variants using a "variants" list.
+MACRO_RENDER_TASKS = [
+    {
+        "template": "banners.html",
+        "macro": "alert_banner",
+        "variants": [
+            {
+                "output": "alert-banner.html",
+                "context": {
+                    "heading": "Default alert banner",
+                    "message": "This is the default alert banner.",
+                },
+            },
+            {
+                "name": "error",
+                "output": "alert-banner-error.html",
+                "context": {
+                    "variant": "error",
+                    "heading": "Error alert banner",
+                    "message": "This is the error alert banner.",
+                },
+            },
+            {
+                "name": "success",
+                "output": "alert-banner-success.html",
+                "context": {
+                    "variant": "success",
+                    "heading": "Success alert banner",
+                    "message": "This is the success alert banner.",
+                },
+            },
+        ],
+    },
+    {
+        "template": "banners.html",
+        "macro": "belly_band",
+        "output": "belly-band.html",
+        "context": {
+            "heading": "Need more help?",
+            "link_text": "Contact support",
+            "link_href": "https://support.example.com",
+        },
+    },
+    {
+        "template": "buttons.html",
+        "macro": "govuk_button",
+        "output": "govuk-button.html",
+        "context": {
+            "text": "Rendered button",
+            "classes": "extra-class",
+        },
+    },
+]
+
+
+def render_macros():
```

Comment:
ooh super interested to pick your brain on this stuff,I was looking into very similar things.

main thing I got blocked on was considering the best way to render our templates; this looks like a great minimal way to get us going.

I am also interested in seeing how we could potentially be more in line with tna and gds, as they use nunjucks but thats js. Would be great if we didnt have to copy and paste the html for their components...